jupyter-extension/jupyterlab_pachyderm/mount_server_client.py

```diff
@@ -114,7 +114,13 @@ async def _ensure_mount_server(self):
                     if os.path.exists(self.mount_dir) or os.path.islink(self.mount_dir):
                         if os.path.isdir(self.mount_dir): 
                             get_logger().debug(f"Removing dir {self.mount_dir}")
-                            os.rmdir(self.mount_dir)
+                            try: 
+                                os.rmdir(self.mount_dir)
+                            except OSError:
```

Comment:
Can we restrict this exception catch in any way since the code in this block is pretty heavy handed? Is the actual error being raised a [`PermissionError`](https://docs.python.org/3/library/exceptions.html#PermissionError)? Or if its a raw `OSError` can we we match the [errno](https://docs.python.org/3/library/errno.html) explicitly?

Reply:
True, but this only triggers a retry.   It will then repeat the removal and if it fails again it will bubble it up just like it would have originally.   So tightening up the check won't really change the resulting visible behavior.

Reply:
I verified that podman at least throws PermisisonError, so tightening up the except clause to that.